instrumentation.go

```diff
@@ -396,3 +408,13 @@ func WithSampler(sampler trace.Sampler) InstrumentationOption {
 		return c, nil
 	})
 }
+
+// WithAdditionalResourceAttributes returns an [InstrumentationOption] that will
+// configure an [Instrumentation] to use the provided attributes as additional
+// OpenTelemetry Resource attributes.
+func WithAdditionalResourceAttributes(attrs []attribute.KeyValue) InstrumentationOption {
+	return fnOpt(func(_ context.Context, c instConfig) (instConfig, error) {
+		c.additionalResAttrs = attrs
+		return c, nil
+	})
+}
```

Comment:
How about making the API similar to https://pkg.go.dev/go.opentelemetry.io/otel/metric#WithAttributes?

Something like:

```suggestion
func WithResourceAttributes(attrs ...attribute.KeyValue) InstrumentationOption {
	return fnOpt(func(_ context.Context, c instConfig) (instConfig, error) {
		c.additionalResAttrs = append(c.additionalResAttrs, attrs...)
		return c, nil
	})
}
```